velox/functions/lib/MapFromEntries.cpp

```diff
@@ -129,8 +132,9 @@ class MapFromEntriesFunction : public exec::VectorFunction {
     });
 
     auto resetSize = [&](vector_size_t row) { mutableSizes[row] = 0; };
+    auto nulls = allocateNulls(decodedRowVector->size(), context.pool());
```

Comment:
This change is a bit convoluted. First, we now allocate nulls buffer unconditionally even when it is not needed / used. Second, there is a confusing logic in LL231 to decide whether to use it or not.

Reply:
Yes. I have to allocate the nulls here since it's need to be updated during following null handling for Spark. The logic in L231 is mostly kind of keeping previous logic for presto. It's a little mixed up here. Do you think it's better to separate the implementation instead of keep it in shared lib?

Reply:
Is it possible to make throwOnNull a tamplate parameter, and provide different implementations base on it? In this way we may can avoid allocating nulls when unnecessary for Presto case.

Reply:
I'm afraid it will make the code more awkward as we still need to allocate nulls in dictionary path. The code will be like `if constexpr (throwOnNull) {allocateNull....}` and after some lines, another `if constexpr (!throwOnNull) {allocateNull....}`.

Reply:
Hi @mbasmanova, in this PR we are attempting to reuse the Presto logic, which appears to cause some implementation confusion. Alternatively, we could add a different Spark implementation while leaving the Presto one untouched. Which option do you think is preferred? Thanks!